code/Controllers/SilverStripeNavigator.php

```diff
@@ -32,6 +36,11 @@ class SilverStripeNavigator extends ViewableData
      */
     public function __construct(CMSPreviewable $record)
     {
+        Deprecation::notice(
```

Comment:
Please wrap this in a call to `Deprecation::withNoReplacement()` since we can't avoid calling this in CMS 4 and don't want to flood logs with it. Same with the other classes.